src/meta/src/manager/metadata.rs

```diff
@@ -0,0 +1,369 @@
+// Copyright 2023 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::{HashMap, HashSet};
+
+use risingwave_common::catalog::{TableId, TableOption};
+use risingwave_pb::catalog::PbSource;
+use risingwave_pb::common::worker_node::{PbResource, State};
+use risingwave_pb::common::{HostAddress, PbWorkerNode, PbWorkerType, WorkerType};
+use risingwave_pb::ddl_service::TableJobType;
+use risingwave_pb::meta::add_worker_node_request::Property as AddNodeProperty;
+use risingwave_pb::meta::table_fragments::Fragment;
+use risingwave_pb::stream_plan::PbStreamActor;
+
+use crate::controller::catalog::CatalogControllerRef;
+use crate::controller::cluster::{ClusterControllerRef, WorkerExtraInfo};
+use crate::manager::{
+    CatalogManagerRef, ClusterManagerRef, FragmentManagerRef, StreamingClusterInfo, WorkerId,
+};
+use crate::model::{ActorId, FragmentId, MetadataModel, TableFragments};
+use crate::MetaResult;
+
+#[derive(Clone)]
+pub enum MetadataManager {
```

Comment:
`MetadataManager` introduced here, providing two metadata managers based on different backend implementations. All RPC services, internal managers should use it instead to acquire and modify metadata.